test/route-test.js

```diff
@@ -1121,6 +1121,29 @@ if (dev) {
 		dev.warn = oldlog;
 	});
 
+	test("should not be display warning for matching keys when the routes do not match (#99)", function () {
+		expect(1);
+		var oldlog = dev.warn;
+
+		dev.warn = function(text) {
+			ok(true, text);
```

Comment:
It might be useful to make this assertion more specific by checking that it contains `"login"` and `"signup"`. Otherwise, you have to look at the assertion message to know for sure the correct check failed.